src/typegoose.ts

```diff
@@ -199,8 +199,13 @@ export function getDiscriminatorModelForClass<T, U extends AnyParamConstructor<T
   if (models.get(name)) {
     return models.get(name) as ReturnModelType<U, T>;
   }
+  const sch = buildSchema(cl) as mongoose.Schema & {paths: object};
+
+  const discriminatorKey =  sch.get('discriminatorKey');
+  if (sch.path(discriminatorKey)) {
+    sch.paths[discriminatorKey].options.$skipDiscriminatorCheck = true;
```

Comment:
*i dont fully understand mongoose, thats why i ask this*

what does "$skipDiscriminatorCheck" do?

Reply:
It is a little bit hacky because it is an internal option normally not used outside of the mongoose lib. But it allows you to override the key of a discriminator class.  This is necessary to support custom discriminatorKeys.

Reply:
after some testing, i couldnt see a difference in the tests, so why was this (whole (in this function)) change needed?

Reply:
I modified the discriminator class to cover this edge case (https://github.com/typegoose/typegoose/pull/21/files#diff-5ea243b73204be263b19204e1f63a933R18) if you remove this line you will see that one test is failing.

Reply:
i mean it worked before, it had the correct discriminator (in the discriminators.ts file), but your *new* test `should set all parent props for nested array items` would fail

Reply:
Okay thanks, for all the testing effort! We are using the discriminator feature in our application heavily and I hade some trouble to support custom discriminator keys. This was my solution to support more complex structures where the discriminator key is a different property than `__t`.  I'm open to any suggestions and enhancements! :)